tools/sync-module-dependencies/README.md

```diff
@@ -0,0 +1,49 @@
+
+# sync-module-dependencies
+
+A small utility to keep Go module replace directives consistent across the repository from a single source of truth.
+
+## What it does
+
+- Reads dependency definitions from the project-level `dependency-replacements.yaml`.
+- Renders the list of `replace` directives using a template.
+- Injects the rendered block into target files (e.g., `go.mod`) between well-known markers.
+- Runs `go mod tidy` for affected modules.
+- Cleans up temporary files created during generation.
+
+<img src="resources/flow_diagram.png" alt="flow diagram" width="200">
```

Comment:
Nit: Could we replace this with a mermaid flowchart (https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-diagrams) so we can maintain it if things evolve